File: 프로그래머스/3/43238. 입국심사/입국심사.java
class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        long left = 0;
        long right = times[times.length-1] * (long)n;

        while(left <= right) {
            long mid = (left + right) / 2;
            long count = 0;

            for (int i = 0; i < times.length; i++){
                count += mid / times[i];
            }

            if (count < n){
                left = mid + 1;
            }
            else {
                right = mid - 1;
                answer = mid;
            }
        }
        System.out.println(answer);
        return answer;
    }
}

File: 프로그래머스/3/43238. 입국심사/입국심사.py
def solution(n, times):
    answer = 0
    left = 0
    right = times[-1] * n

    while left <= right:
        mid = (left + right) // 2
        count = 0

        for time in times:
            count += mid // time

        if count < n:
            left = mid + 1
        else:
            right = mid - 1
            answer = mid

    print(answer)
    return answer